deepchecks/nlp/checks/data_integrity/under_annotated_segments.py

```diff
@@ -0,0 +1,290 @@
+
+            # Add box trace to the legend using lines
+            fig.add_trace(go.Scatter(
+                x=[feature_1_lower, feature_1_lower, feature_1_upper, feature_1_upper, feature_1_lower],
+                y=[feature_2_lower, feature_2_upper, feature_2_upper, feature_2_lower, feature_2_lower],
+                mode='lines',
+                line=dict(color='rgb(121, 100, 255)', width=3, dash='dot'),
+                name='Under Annotated Segment',
+                fill='toself', fillcolor='rgb(121, 100, 255)', opacity=0.4,
+                legendgroup='box'))
+
+            # Add annotated scatter plot
+            fig.add_trace(go.Scatter(x=annotated_data[feature_1], y=annotated_data[feature_2],
+                                     mode='markers', opacity=0.7,
+                                     marker=dict(symbol='circle', color='lightgreen', size=10,
+                                                 line=dict(color='black', width=1)),
+                                     hovertemplate=hover_template + 'True',
+                                     text=annotated_data['text'],
+                                     name='Annotated Samples'))
+
+            # Add not annotated scatter plot
+            fig.add_trace(go.Scatter(x=not_annotated_data[feature_1], y=not_annotated_data[feature_2],
+                                     mode='markers', opacity=0.7,
+                                     marker=dict(symbol='x', color='red', size=10, line=dict(color='black', width=1)),
+                                     hovertemplate=hover_template + 'False',
+                                     text=not_annotated_data['text'],
+                                     name='Not Annotated Samples'))
+
+            # Update figure layout
+            fig.update_layout(title=dict(
+                text=f'Under Annotated Segment ({row["% of Data"]}% of Data)<br><sub>Annotation ratio in segment ' \
+                     f'is {format_percent(row["Annotation Ratio"])} (vs. {format_percent(avg_score)}'
+                     f' in whole data)</sub>',
+                font=dict(size=24)),
+                xaxis_title=feature_1, yaxis_title=feature_2 if feature_2 != 'virtual_col' else '',
+                autosize=False, width=1000, height=600,
+                font=dict(size=14),
+                plot_bgcolor='rgba(245, 245, 245, 1)',
+                xaxis=dict(gridcolor='rgba(200, 200, 200, 0.5)',
+                           zerolinecolor='rgba(200, 200, 200, 0.5)'),
+                yaxis=dict(gridcolor='rgba(200, 200, 200, 0.5)',
+                           zerolinecolor='rgba(200, 200, 200, 0.5)'))
+
+            display_tabs[tab_title] = [fig, f'Check ran on {encoded_data.shape[0]} data samples.<br>{msg}']
+
+        return DisplayMap(display_tabs)
+
+    def add_condition_segments_annotation_ratio_greater_than(self, threshold: float = 0.70):
```

Comment:
Add condition for "segment is less annotated than the avg annotation ratio by x"

Reply:
there is from abstract class